03-Modify-Raster.Rmd

```diff
@@ -74,197 +72,240 @@ Read more about the `raster` package in R.</a>
+2. The rasters are stored as individual files. 
+3. The computations needed are complex. 
 
-Let's have a look at how the `overlay` function works. The `overlay` function 
-takes two or more rasters and applies a function to them using efficient processing
-methods. The syntax looks like:
+>NOTE: If the rasters are stacked and stored as `RasterStack` or `RasterBrick`
+>objects in R, then we should use `calc()`. `overlay()` will not work on stacked
+>rasters. 
 
-`outputRaster <- overlay(raster1, raster2, functionName )`
+The `overlay()` function takes two or more rasters and applies a function to
+them using efficient processing methods. The syntax is
 
-##Writing Functions in R
-
-A function represents a set of tasks that need to be repeated over and over. 
-The syntax for a function in R is as follows:
+`outputRaster <- overlay(raster1, raster2, fun=functionName)`
 
-`functionName <- function(variable1, variable2){
+##Writing Functions in R
+Before we can continue with `overlay()` we need to discuss writing functions. A 
+function represents a set of tasks that need to be repeated over and over. 
 
+LINK TO MORE ABOUT FUNCTIONS IN R 
+I like explanations (how to AND why) here: http://nicercode.github.io/guides/functions/
+but I'm hesitant to link to blogs that might disappear so also QuickR link:
+http://www.statmethods.net/management/userfunctions.html
+Thoughts? 
 
-}
+A simplified syntax for a function in R is as follows:
 
-LINK TO MORE ABOUT FUNCTIONS IN R 
+`functionName <- function(variable1, variable2){WhatYouWantDone, WhatToReturn}`
 
+Let's say we want to create a function that subtracts one raster from another. 
+In our case what we want does is what we want returned so we don't have to 
+specify it twice. 
 ```{r create-function }
-
-#first, let's create a function
 #this function will take two rasters (r1 and r2) and subtract them
 subtRasters <- function(r1, r2){
-return(r1-r2)
+(r1-r2)
 }
-
 ```
 
+We now have a function called `subtRasters` that will subtract any two rasters
+that we give it.  
 
+##Continuing with Overlay Function
+Let's repeat our simple raster math computation using `overlay()` and the
+function we just created. 
 ```{r raster-overlay }
+CHM_ov<- overlay(DSM,DTM,fun=subtRasters)
 
-
-CHM_ov_HARV <- overlay(DSM,DTM,fun=subtRasters)
-
-plot(CHM_ov_HARV,
+plot(CHM_ov,
      main="NEON Canopy Height Model - Overlay Subtract\n Harvard Forest")
-
 ```
 
-So now, we've learned two different ways to subtract two rasters in R.
-Once we have an output raster, we can use the `writeRaster` function to export
-the object as a new geotiff. 
+#Writing Raster Files
+So now we've learned two different ways to do calculations (in our case 
+subtraction) with two rasters in R.  This gives us an output raster object 
+(`CHM_ov`), we can use the `writeRaster` function to export the object as a new 
+geotiff. 
+
+>NAMES: Part of writing a raster (or any file) is choosing a suitable name for 
```

Comment:
Hi @mjones01 i think we need the _HARV throughout. IT wasn't consistent because i added the SJER data after but i will have them use SJER data in challenges. So, let's keep the naming consistent with HARV at the end for all HARV data. thank you!